api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -884,4 +897,278 @@ public ConversationWebhookList listConversationWebhooks() throws UnauthorizedExc
+        return messageBirdService.requestList(url, new Paging.PagedPaging(page, pageSize), VoiceCallResponseList.class);
+    }
+
+    /**
+     * Function to view voice call by id
+     *
+     * @param id Voice call ID
+     * @return VoiceCallResponse
+     * @throws UnauthorizedException
+     * @throws GeneralException
+     */
+    public VoiceCallResponse viewVoiceCall(final String id) throws NotFoundException, GeneralException, UnauthorizedException {
+        if (id == null) {
+            throw new IllegalArgumentException("Voice Message ID must be specified.");
+        }
+
+        String url = VOICE_CALLS_BASE_URL + VOICECALLSPATH;
+        return messageBirdService.requestByID(url, id, VoiceCallResponse.class);
+    }
+
+    /**
+     * Function to delete voice call by id
+     *
+     * @param id Voice call ID
+     * @return
+     * @throws UnauthorizedException
+     * @throws GeneralException
+     */
+    public void deleteVoiceCall(final String id) throws NotFoundException, GeneralException, UnauthorizedException {
+        if (id == null) {
+            throw new IllegalArgumentException("Voice Message ID must be specified.");
+        }
+
+        String url = VOICE_CALLS_BASE_URL + VOICECALLSPATH;
+        messageBirdService.deleteByID(url, id);
+    }
+
+    /**
+     * Retrieves a listing of all legs.
+     *
+     * @param callId   Voice call ID
+     * @param page     page to fetch (can be null - will return first page), number of first page is 1
+     * @param pageSize page size
+     * @return
+     * @throws UnsupportedEncodingException
+     * @throws UnauthorizedException
+     * @throws GeneralException
+     */
+    public VoiceCallLegResponse viewCallLegsByCallId(String callId, Integer page, Integer pageSize) throws UnsupportedEncodingException, UnauthorizedException, GeneralException {
+        String url = VOICE_CALLS_BASE_URL + VOICECALLSPATH + '/' + urlEncode(callId) + VOICELEGS_SUFFIX_PATH;
+
+        return messageBirdService.requestList(url, new Paging.PagedPaging(page, pageSize), VoiceCallLegResponse.class);
+    }
+
+    /**
+     * Retrieves a leg resource.
+     * The parameters are the unique ID of the call and of the leg that were returned upon their respective creation.
+     *
+     * @param callId Voice call ID
+     * @param legId  ID of leg of specified call {callId}
+     * @return
+     * @throws UnsupportedEncodingException
+     * @throws NotFoundException
+     * @throws GeneralException
+     * @throws UnauthorizedException
+     */
+    public VoiceCallLeg viewCallLegByCallIdAndLegId(final String callId, String legId) throws UnsupportedEncodingException, NotFoundException, GeneralException, UnauthorizedException {
+        String url = VOICE_CALLS_BASE_URL +
+                VOICECALLSPATH + "/" + urlEncode(callId) +
+                VOICELEGS_SUFFIX_PATH;
+
+        VoiceCallLegResponse response = messageBirdService.requestByID(url, legId, VoiceCallLegResponse.class);
```

Comment:
We could probably abstract this away from checking if data has anything to adding a method to VoiceCallLegResponse that can tell you if there data.